syntax = "proto3";        //说明使用proto3语法定义协议

package nearline_engine;
service NearlineEngine{
  rpc OLAPRequest (OLAPRequestInfo) returns(Response) {}
  rpc TextEmbeddingRequest (TextEmbeddingRequestInfo) returns (Response) {}
}

message RequestBaseInfo{
  string pipeline_id = 1;                      //请求
  string config = 2;                           //请求样本信息
}

message OLAPRequestInfo{
  RequestBaseInfo request_base_info = 1 ;
}

message TextEmbeddingRequestInfo{
  RequestBaseInfo request_base_info = 1 ;
}

message Response{
  int32 response_info = 1;                    //返回信息,0是正确
}